UPGRADING.md

```
@@ -14,5 +19,6 @@ As a result, MQT YAQS is no longer tested under Python 3.9 and requires Python 3
 
 <!-- Version links -->
 
-[unreleased]: https://github.com/munich-quantum-toolkit/yaqs/compare/v0.3.2...HEAD
+[Unreleased]: https://github.com/munich-quantum-toolkit/yaqs/compare/v0.3.3...HEAD
+[0.3.3]: https://github.com/munich-quantum-toolkit/yaqs/releases/tag/v0.3.3
```

markdownlint-cli2 (0.20.0)

[warning] Link and image reference definitions should be needed
Unused link or image reference definition: "0.3.3"

(MD053, link-image-reference-definitions)